pilot/pkg/model/credentials/resource.go

```diff
@@ -54,6 +54,10 @@ func (sr SecretResource) Key() string {
 	return "sds://" + sr.Type + "/" + sr.Name + "/" + sr.Namespace + "/" + string(sr.Cluster)
 }
 
+func (sr SecretResource) KubernetesResourceName() string {
```

Comment:
This seems a bit awkward since `Type` could be something other than `KubernetesSecretType`. This seems like it _could_ lead to type-confusion bugs

Reply:
Updated to use `sr.Type` for type.